### PR TITLE
Overlay Walkthrough

### PR DESCRIPTION
**Pull request by azhara-a**
Created a tutorial on how Recipe Finder works. 

The tutorial is available on the Login and Sign Up pages.

**Video demo of the tutorial:** https://drive.google.com/file/d/1bcreZrs6NmNH3zqp6YYYVcbWqD3dOKV2/view?usp=sharing

**Login Page**
Screenshot 2020-09-15 at 12 04 14 PM

**Sign Up Page**
image

**One of the pages**
image

### CODE REVIEW

File: frontend/src/components/recommended-recipes/RecommendedRecipes.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ class RecommendedRecipes extends Component {
     }
 
     if (this.state.isLoading) {
-      return loading("Scanning recipes ...");
+      return loading("Scanning recipes...", /** withBackground*/ true);
```
Comment (mam249) [inline frontend/src/components/recommended-recipes/RecommendedRecipes.js:70]: what does the /** withBackground*/ do

**Review by mam249 — COMMENTED** [review]
one bit of confusion in RecommendedRecipes.js but otherwise LGTM

File: frontend/src/components/recommended-recipes/RecommendedRecipes.js (hunk 1 above)
Comment (azhara-a) [reply frontend/src/components/recommended-recipes/RecommendedRecipes.js:70]: See function loading below. If no text or withBackground are provided, the function will use the default values which are "Loading ..." and display loading without any background. 

For most requests to the backend, it doesn't take that long to get the response back, so for pages such as Favourites or History that have white background, the loading with a background might look like a glitch. That is why I created this withBackground variable with default value false as most pages have white background.

Whereas for scanning recipes, it takes quite a while to load the recipes from the backend so a background picture looks nicer.
```
export function loading(text = "Loading...", withBackground = false) {
  const background = withBackground ? getBackground() : "";
  return (
    <div>
      {background}
      <div className="spinner-div">
        <div className="spinner-border" role="status">
          <span className="sr-only">Loading...</span>
        </div>
        <h3>{text}</h3>
      </div>
    </div>
  );
}
```

**Review by lminchillo — APPROVED** [review]
LGTM!

File: frontend/src/components/recommended-recipes/RecommendedRecipes.js (hunk 1 above)
Comment (mugdhalakhani) [reply frontend/src/components/recommended-recipes/RecommendedRecipes.js:70]: One thought here: Make sure to test your website in dark mode or provide a dark mode theme. Can be done after the main presentation.

**Review by mugdhalakhani — COMMENTED** [review]
I haven't gone through all changes yet, but after our discussion yesterday, I think you were going to make these changes:
1. Make the walkthrough shorter (3-4 screens)
2. Remove the popup on login if the user hasn't taken the tour
3. Add a link/button to launch the tour once logged in.

Were you planning to make those changes later? I think 1 can be done here?

**Comment by azhara-a** [comment]
2. The popup on login is removed, now it's just a toast at the top of sign up with a question: "Do you want to see how Recipe Finder works?"
3. The button to launch the tour is "Yes" on the toast". Once the user leaves the tour, the toast disappears as well
> image

**Comment by mugdhalakhani** [comment]
Thanks Azhara!

> 3\. The button to launch the tour is "Yes" on the toast". Once the user leaves the tour, the toast disappears as well
I don't dislike the Toast idea but there are two issues with this as is:
1. The user won't be able to access the tour if they didn't click yes on the toast before it times out. Consider having a button or a link in the header so the user can access the tour anytime.
2. When the user is in the tour, they don't need to see the toast, it can disappear as soon as the user clicks yes.

WDYT? I'll review the rest of this now.

**Comment by azhara-a** [comment]
> 1. The user won't be able to access the tour if they didn't click yes on the toast before it times out. Consider having a button or a link in the header so the user can access the tour anytime.

The toast doesn't time out on the sign up page, however, the user can't access the tour once they enter the site. I will add the tour button in the header to accommodate users who are logged in and still want to see the tour.

> 2. When the user is in the tour, they don't need to see the toast, it can disappear as soon as the user clicks yes.

That already happens

**Review by mugdhalakhani — APPROVED** [review]

**Pull request merged**